src/emulator/auth/operations.ts

```diff
@@ -1460,7 +1460,43 @@ function signInWithIdp(
     }
   }
 
-  let { response, rawId } = fakeFetchUserInfoFromIdp(providerId, claims);
+  if (reqBody.pendingToken) {
+    const decoded = decodeJwt(reqBody.pendingToken, { complete: true }) as { expiresAt: number };
+    assert(decoded.expiresAt < Date.now(), "INVALID_PENDING_TOKEN");
+  }
+
+  const encryptedSessionState =
+    normalizedUri.searchParams.get("RelayState") ||
+    normalizedUri.searchParams.get("state") ||
+    undefined;
+  let sessionState = undefined;
+  if (encryptedSessionState) {
+    const sessionStateJson = Buffer.from(encryptedSessionState, "base64").toString("utf8");
+    sessionState = JSON.parse(sessionStateJson) as SessionState;
+    assert(sessionState, "MISSING_SESSION_ID");
+    assert(sessionState.sessionId === reqBody.sessionId, "INVALID_SESSION_ID");
+  }
+
+  // Generic SAML flow
+  let samlResponse: SamlResponse | undefined;
+  if (normalizedUri.searchParams.get("SAMLResponse")) {
+    // Auth emulator purposefully does not parse SAML and expects SAML-related
+    // fields to be JSON objects.
+    samlResponse = JSON.parse(normalizedUri.searchParams.get("SAMLResponse")!) as SamlResponse;
```

Comment:
This seems reasonable. Can we also populate this param in the mock widget? Like set `nameId` equal to user-input `email`?